### PR TITLE
Attribute-based reverse mapping with SourceMemberAttribute

### LINKED ISSUE #3129: Attribute-based Reverse Mapping with SourceMember does not work
### Source/destination types

```csharp
    public class Company
    {
        public int Id { get; set; }
        public string Name { get; set; }
    }

    [AutoMap(typeof(Company), ReverseMap = true)]
    public class ViewModel
    {
        public int Id { get; set; }
        [SourceMember(nameof(Company.Name))]
        public string CompanyName { get; set; }
    }
```

### Mapping configuration

```csharp
services.AddAutoMapper(Assembly.GetExecutingAssembly());
```

### Versions
    <PackageReference Include="AutoMapper" Version="8.1.1" />
    <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="6.1.1" />

### Expected behavior

Reverse mapping of a property attributed with `SourceMember` should also map the property.

### Actual behavior

Mapping does not work.

### Steps to reproduce

A full repro repository is found here: [AutoMapperAttributeMappingIssue](https://github.com/dradovic/AutoMapperAttributeMappingIssue)

### PR DESCRIPTION
**Pull request by timmi-on-rails**
Based on PR #3336
Added test and fix.

Fixes #3129 .
#3329

### CODE REVIEW

File: src/UnitTests/AttributeBasedMaps.cs [hunk 1]
```diff
@@ -77,6 +77,82 @@ public void Should_validate_successfully()
             }
         }
 
+        public class When_specifying_map_and_reverse_map_with_attribute_and_sourcemember_attribute : NonValidatingSpecBase
```
Comment (lbargaoanu) [inline src/UnitTests/AttributeBasedMaps.cs]: This needs to be shorter :)

File: src/UnitTests/AttributeBasedMaps.cs [hunk 2]
```diff
@@ -77,6 +77,82 @@ public void Should_validate_successfully()
...
+
+            [Fact]
+            public void Should_validate_successfully()
+            {
+                typeof(AutoMapperConfigurationException).ShouldNotBeThrownBy(() => Configuration.AssertConfigurationIsValid(Configuration.FindTypeMapFor<Source, Destination>()));
+                typeof(AutoMapperConfigurationException).ShouldNotBeThrownBy(() => Configuration.AssertConfigurationIsValid(Configuration.FindTypeMapFor<Destination, Source>()));
+            }
+        }
+
+        public class When_specifying_open_generic_map_and_reverse_map_with_attribute_and_sourcemember_attribute : NonValidatingSpecBase
```
Comment (lbargaoanu) [inline src/UnitTests/AttributeBasedMaps.cs]: Again, too long.

**Comment by timmi-on-rails** [comment]
Can you please trigger CI again. There seems to be an issue not related to the sourcecode

**Comment by lbargaoanu** [comment]
It's ok, it will run again when you rebase after that other PR is merged.

**Comment by lbargaoanu** [comment]
@timmi-on-rails Can you please rebase on master?

**Comment by timmi-on-rails** [comment]
I got a conflict, when I wanted to rebase on master.
Can you work with this here?

**Comment by lbargaoanu** [comment]
It's all your code. Surely you can solve any conflicts with your own code :)

**Comment by timmi-on-rails** [comment]
Ok, I think I got it right now.

Doesn't rebasing and force pushing invalidate your code reviews, because I could make a mistake by accident (when resolving conflicts)?
I am inexperienced in rebasing across forks and extended git usage.

**Comment by lbargaoanu** [comment]
Yes, the idea is to rebase and review again. But all I'm looking the second time around is rebase errors, so it's quick. Thanks for taking the time to do everything properly. I'm sure it'll be easier next time :)

**Pull request merged**